### PR TITLE
Python 3 UIA in consoles: use textUtils for word offsets

### PR DESCRIPTION
**Pull request by LeonarddeR**
### Link to issue number:
None

### Summary of the issue:
The UIA implementation for windows consoles uses uniscribe to calculate word boundaries. However, this code didn't take wide string and python 3 string differences into account. I'm actually pretty sure that the code was also broken in Python 2, as it calculated an offset based on per character movement, which was likely to take surrogate pairs into account, while uniscribe doesn't.

### Description of how this pull request fixes the issue:
Use textUtils in this code in a similar way as in OffsetsTextinfo.

### Testing performed:
t.b.d. I currently have an insider build on my system which crashes command prompt as soon as I paste in surrogate characters.

### Known issues with pull request:
None

### Change log entry:
None

### CODE REVIEW

**Comment by leonardder** [comment]
Cc @codeofdusk 

Note that it *does not* fix #9875

File: source/NVDAObjects/UIA/winConsoleUIA.py [hunk 1]
```diff
@@ -192,16 +193,22 @@ def _getWordOffsetsInThisLine(self, offset, lineInfo):
 		# not more than two alphanumeric chars in a row.
 		# Inject two alphanumeric characters at the end to fix this.
 		lineText += "xx"
```
Comment (feerrenrut) [inline source/NVDAObjects/UIA/winConsoleUIA.py]: Is this still necessary?

File: source/NVDAObjects/UIA/winConsoleUIA.py [hunk 2]
```diff
@@ -192,16 +193,22 @@ def _getWordOffsetsInThisLine(self, offset, lineInfo):
...
-		)
+		):
+			log.error("Calculating word offsets with uniscribe failed")
+			return (offset, offset + 1)
+		wordStart, wordEnd = offsetConverter.wideToStrOffsets(start.value, end.value)
 		return (
-			start.value,
-			min(end.value, max(1, len(lineText) - 2))
+			wordStart,
+			min(wordEnd, max(1, len(lineText) - 2))
```
Comment (feerrenrut) [inline source/NVDAObjects/UIA/winConsoleUIA.py]: Is `len(lineText)` actually right here?

File: source/NVDAObjects/UIA/winConsoleUIA.py (hunk 1 above)
Comment (leonardder) [reply source/NVDAObjects/UIA/winConsoleUIA.py]: I do think so, yes. Probably hard to verify.

File: source/NVDAObjects/UIA/winConsoleUIA.py (hunk 2 above)
Comment (leonardder) [reply source/NVDAObjects/UIA/winConsoleUIA.py]: Yes. These offsets are converted to python 3 style string offsets.

**Comment by leonardder** [comment]
Actually, this was the first time I was actually able to test this, and it looks like UIA is treating surrogates as stand alone characters in the console, at least in bash. This means that both the current and the previous code is likely to fail.

**Comment by feerrenrut** [comment]
> This means that both the current and the previous code is likely to fail.

Am I interpreting correctly that this PR is not ready?

**Comment by leonardder** [comment]
Yes. The current code is safe as soon as there are no emoji in the console.

**Comment by feerrenrut** [comment]
In that case, I'll close this PR. If we have a need for this approach we can always re-open it.

**Pull request closed without merging**